File: bash/code_to_gast/bash_router.py
import bashlex
import bash.code_to_gast.bash_helpers as bash_helpers

def node_to_gast(node):
   if type(node) == list:
      node_list = []
      for part in node:
         node_list.append(node_to_gast(part))
      return node_list
   elif node.kind == 'command':
      return node_to_gast(node.parts)
   elif node.kind == 'word':
      return bash_helpers.word_to_gast(node)

File: bash/code_to_gast/bash_router.js
const bash_helpers = require('./bash_helpers.js')

function node_to_gast(node) {
    if (Array.isArray(node)) {
        return bash_helpers.array_to_gast(node)
    } else if (node["type"] === 'Script') {
        return bash_helpers.script_to_gast(node)
    } else if (node["type"] === 'Command') {
        return bash_helpers.command_to_gast(node)
    } else if (node["type"] === 'Word') {
        return bash_helpers.word_to_gast(node)
    }
}

exports.node_to_gast = node_to_gast

